src/lib/comms/include/sol-netctl.h

```diff
@@ -553,6 +614,73 @@ sol_netctl_find_service_by_name(const char *service_name)
 }
 
 /**
+ * @brief register a agent for network connection
+ *
+ * A single agent is registered for an application that registering a new agent.
+ *
+ * @see struct sol_netctl_agent
+ *
+ * @param agent the agent Callback struct to be called when the related information is updated.
+ * @param data Add a user data per callback.
+ *
+ * @return 0 on success, -errno on failure.
+ */
+int sol_netctl_register_agent(const struct sol_netctl_agent *agent, const void *data);
+
+/**
+ * @brief unregister a agent for network connection
+ *
+ * unregister the previous and to unregister the last agent you should use it.
```

Comment:
it's very confusing this phrase. To my mind you don't need this, the brief is good enough.